ravenframework/Optimizers/RavenSampled.py

```diff
@@ -313,57 +321,140 @@ def finalizeSampler(self, failedRuns):
       @ In, failedRuns, list, runs that failed as part of this sampling
       @ Out, None
     """
-    # get and print the best trajectory obtained
-    bestValue = None
-    bestTraj = None
-    bestPoint = None
-    s = -1 if self._minMax == 'max' else 1
-    # check converged trajectories
-    self.raiseAMessage('*' * 80)
-    self.raiseAMessage('Optimizer Final Results:')
-    self.raiseADebug('')
-    self.raiseADebug(' - Trajectory Results:')
-    self.raiseADebug('  TRAJ   STATUS    VALUE')
-    statusTemplate = '   {traj:2d}  {status:^11s}  {val: 1.3e}'
-    # print cancelled traj
-    for traj, info in self._cancelledTraj.items():
-      val = info['value']
-      status = info['reason']
-      self.raiseADebug(statusTemplate.format(status=status, traj=traj, val=s * val))
-    # check converged traj
-    for traj, info in self._convergedTraj.items():
+    if not self._canHandleMultiObjective or len(self._objectiveVar) == 1:
+      # get and print the best trajectory obtained
+      bestValue = None
+      bestTraj = None
+      bestPoint = None
+      s = -1 if 'max' in self._minMax else 1
+      # check converged trajectories
+      self.raiseAMessage('*' * 80)
+      self.raiseAMessage('Optimizer Final Results:')
+      self.raiseADebug('')
+      self.raiseADebug(' - Trajectory Results:')
+      self.raiseADebug('  TRAJ   STATUS    VALUE')
+      statusTemplate = '   {traj:2d}  {status:^11s}  {val: 1.3e}'
+      # print cancelled traj
+      for traj, info in self._cancelledTraj.items():
+        val = info['value']
+        status = info['reason']
+        self.raiseADebug(statusTemplate.format(status=status, traj=traj, val=s * val))
+      # check converged traj
+      for traj, info in self._convergedTraj.items():
+        opt = self._optPointHistory[traj][-1][0]
+        val = info['value']
+        self.raiseADebug(statusTemplate.format(status='converged', traj=traj, val=s * val))
+        if bestValue is None or val < bestValue:
+          bestTraj = traj
+          bestValue = val
+      # further check active unfinished trajectories
+      # FIXME why should there be any active, unfinished trajectories when we're cleaning up sampler?
+      traj = 0 # FIXME why only 0?? what if it's other trajectories that are active and unfinished?
+      # sanity check: if there's no history (we never got any answers) then report rather than crash
+      if len(self._optPointHistory[traj]) == 0:
+        self.raiseAnError(RuntimeError, f'There is no optimization history for traj {traj}! ' +
+                          'Perhaps the Model failed?')
       opt = self._optPointHistory[traj][-1][0]
-      val = info['value']
-      self.raiseADebug(statusTemplate.format(status='converged', traj=traj, val=s * val))
+      val = opt[self._objectiveVar[0]]
+      self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
       if bestValue is None or val < bestValue:
```

Comment:
@JunyungKim, we have 'first', 'accepted' and 'converged', maybe 'final' but are we adding 'active'? how is this different than 'accepted' or is it a replacement of 'accepted'. If so, shouldn't this be in Genetic algorithm.py so you do not introduce changes to GradientDescent?